src/consensus/encode.rs

```diff
@@ -673,6 +673,13 @@ fn read_bytes_from_finite_reader<D: io::Read>(mut d: D, mut opts: ReadBytesFromF
     Ok(ret)
 }
 
+impl Encodable for [u8] {
```

Comment:
I would avoid doing this. It makes it quite error-prone, because it's unclear if slices would be length-prefixed or not. If we use this internally, I'd prefer using `consensus_encode_with_size` instead..

Reply:
Generally, we just use the `emit_slice` method from `WriteExt`.

Reply:
Good point, though IIRC `Vec<u8>` implements `Encodable` and it shouldn't for the same reason.